sdks/java/core/src/main/java/org/apache/beam/sdk/io/LocalFileSystem.java

```diff
@@ -80,8 +82,20 @@
 class LocalFileSystem extends FileSystem<LocalResourceId> {
 
   private static final Logger LOG = LoggerFactory.getLogger(LocalFileSystem.class);
+  private static final CopyOption[] DEFAULT_RENAME_COPY_OPTIONS =
+      new CopyOption[] {StandardCopyOption.ATOMIC_MOVE};
 
-  LocalFileSystem() {}
+  private final CopyOption[] renameCopyOption;
+
+  LocalFileSystem() {
+    renameCopyOption = DEFAULT_RENAME_COPY_OPTIONS;
+  }
+
+  // Exists to allow testing behaviour of the FileSystems utility methods
+  @VisibleForTesting
+  LocalFileSystem(CopyOption[] renameCopyOption) {
```

Comment:
Do you expect users to use other options. Also, not sure how this can be used since users usually don't directly interact with LocalFileSystem class.

Reply:
No - this exists only so we can construct a LocalFileSystem to behave like other FS for use in the FileSystems tests.

Reply:
The original tests for FileSystem already test the behavior of LocalFileSystem with atomic move. The other FileSystem implementations like HDFS do not have that behavior though. 

I considered that to test FileSystems.rename we could either:
1) Use a LocalFileSystem configured in a way to behave like other FS implementations (of course you would not use it in that manner normally)
2) Have a standard test suite for FileSystems but run it in the tests of each FileSystem implementation

I opted for approach 1) here but I get the impression that is not particularly clear so perhaps 2) would be better after all?

Reply:
Thanks for the clarification. I agree that (2) is better but we cannot do this in unittests unless we have Good fakes for file-systems. We should increase integration coverage for this though but can be a separate PR.  So I'm fine with going for (1) here.